scipy/sparse/linalg/_isolve/tests/test_lgmres.py

```diff
@@ -53,11 +58,13 @@ def test_preconditioner(self):
         M = LinearOperator(matvec=pc.solve, shape=A.shape, dtype=A.dtype)
 
         x0, count_0 = do_solve()
-        x1, count_1 = do_solve(M=M)
+        niter[0] = 0
+        x1, count_1 = do_solve(M=M, callback=cb)
 
         assert_(count_1 == 3)
         assert_(count_1 < count_0/2)
         assert_(allclose(x1, x0, rtol=1e-14))
+        assert_(niter[0] < 3)
```

Comment:
Shall we start using the modern/plain `pytest` `assert` for new cases?

One other question here, because it isn't clear to me from the description--is the purpose here to ensure that `do_solve()` finishes in 3 iterations?  Similar to limiting with `maxiter` set to `3`?